lib/datadog/error_tracking/filters.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Datadog
+  module ErrorTracking
+    # Based on configuration, the tracepoint listening to :rescue or :raise
+    # may report more handled errors than we want to report. Therefore we need
+    # a function to filter the events. As the filter function both depends
+    # on configuration and is called numerous time, we generate it during
+    # during the initialization of the feature to have the best performance
+    # possible.
+    #
+    # @api private
+    module Filters
+      module_function
+
+      def get_gem_name(file_path)
+        regex = %r{gems/([^/]+)-\d}
+        regex_match = regex.match(file_path)
+        return unless regex_match
+
+        gem_name = regex_match[1]
+
+        begin
+          Gem::Specification.find_by_name(gem_name)
+        rescue
+          nil
+        end
+      end
+
+      def user_code?(file_path)
+        !get_gem_name(file_path)
+      end
+
+      def datadog_code?(file_path)
+        file_path.include?('lib/datadog/')
+      end
+
+      def third_party_code?(file_path)
+        get_gem_name(file_path) && !datadog_code?(file_path)
+      end
+
+      def file_included?(file_path, instrumented_files)
+        instrumented_files.include?(file_path)
+      end
+
+      # Generate the proc used in the tracepoint
+      def generate_filter(to_instrument_scope, handled_errors_include = nil)
+        case to_instrument_scope
```

Comment:
instead of return in each `when` the last statement should go to else, and `return`-s could be eliminated

Reply:
addressed in e5fe33bee6f4993593a80e133f14416eee497861